feature:sway

saylesss88 opened the pull request:
- **feat: Add sway configuration and SDDM option**
- **feat: sway window manager initial PR**

Initial Rough draft of Sway

saylesss88 commented on nixos/modules/programs/gui/sway/Wallpapers/wallpaper-theme-converter24.png:
I just added a few extra wallpapers for you.

saylesss88 commented on nixos/modules/programs/gui/sway/fuzzel.nix:
```diff
@@ -0,0 +1,10 @@
+    enable = true;
+    settings = {
+      main = {
+        terminal = "${pkgs.ghostty}/bin/ghostty";
```
Yeah, I figured. I just wanted to get a working configuration. This is pieces of my old sway config that I made work for you. We'll adjust it to your liking.

saylesss88 commented on nixos/modules/programs/gui/sway/waybar.nix:
Yeah, I currently have a better config for my waybar on hyprland I'll share with you for ideas. Again the idea was to just share what I had and get your feedback and also to see if you got it to build successfully? That was more what I wanted to know.

saylesss88 commented on nixos/modules/programs/gui/sway/wlogout.nix:
```diff
@@ -0,0 +1,101 @@
+    ];
+    style = ''
+      * {
+        font-family: "JetBrainsMono NF", FontAwesome, sans-serif;
```
Yes, we can use stylix. That's actually what I started with but moved away from it because it injects it's special stylix settings everywhere and doesn't tell you where they are. The auto coloring is cool but it forces them to inject things into your files all over the place.

GregHilston replied:
Gotcha. i look to you for guidance on this then

saylesss88 commented:
Cool, yeah we'll get it to where you want it. I was able to get it to build on my intel laptop successfully and it was functional but needs a lot of work haha. What I did was change the isengard configuration to my host and username and generated my own hardware-configuration.nix and built it. FYI my intel laptop is painfully slow haha only 4G of RAM but it works..

saylesss88 commented:
I have made quite a few changes to this taking your feedback. 

- Updated the waybar config
- Enabled stylix for the sway config
- Fixed the swaylock and swayidle configs
- Switched the shell to alacritty
- Got rid of the `/nix/store/` path hardcoded in
- There's still more to do, let me know if this being a draft affects you being able to test it

GregHilston commented:
The draft status has zero impact on my ability to test this out ;)
